R/util-checkSpec.R

```diff
@@ -0,0 +1,66 @@
+# Check that all data.frames and columns in the spec are present in the data
+#' Function to check if the data and spec are compatible
+#'
+#' This function checks if the data and spec are compatible by comparing the data.frames
+#' and columns in the spec with the data.
+#'
+#' @param lData A list of data.frames.
+#' @param lSpec A list s specifying the expected structure of the data.
```

Comment:
Would it be useful to have column data type in this `lSpec` list as well? This would allow us to confirm not just structure but also ensure that the content is what we expect. could be an optional parameter. So, when used, it would look something like:
```
Raw_QUERY:
  subjectname: 
     required: true
     datatype: character
 querystatus: 
     required: true
     datatype: character
  queryage:
     required: true
     datatype: integer
```

Reply:
I like this better as well, and the current implication only considers the names anyway so it works.

Reply:
Yep, I'm ok with the slightly more complex implementation. Will update.